### PR TITLE
Code Execution requests

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/core/node/new/credentials.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def __eq__(self, other: Any) -> bool:
         return self.verify_key == other.verify_key
 
     def __repr__(self) -> str:
-        return str(self)
+        return "..." + str(self)[-5::]
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/credentials.py:46]: This is the public key so there is no need to hide it.

File: packages/syft/src/syft/core/node/new/task/oblv_service.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    if deployment_id in OBLV_PROCESS_CACHE:
+        process = OBLV_PROCESS_CACHE[deployment_id][0]
+        if process.poll() is None:
+            return process
+        # If the process has been terminated create a new connection
+        del OBLV_PROCESS_CACHE[deployment_id]
+
+    # Always create key file each time, which ensures consistency when there is key change in database
+    create_keys_from_db(oblv_keys_stash)
+    key_path = os.getenv("OBLV_KEY_PATH", "/app/content")
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/task/oblv_service.py:89]: Is is possible to run these code paths without the final oblivious container? Should we not have a different default OBLV_KEY_PATH for testing?

File: packages/syft/src/syft/core/node/new/task/oblv_service.py [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+from .util import find_available_port
+
+# caches the connection to Enclave using the deployment ID
+OBLV_PROCESS_CACHE: Dict[str, List] = {}
+
+
+# TODO: 🟡 Duplication of PyPrimitive Dict
+# This is emulated since the action store curently accepts  only SyftObject types
+@serializable(recursive_serde=True)
+class DictObject(SyftObject):
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/task/oblv_service.py:56]: Can this just be an AnyActionObject?

File: packages/syft/src/syft/core/node/new/task/oblv_service.py [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+        context: AuthedServiceContext,
+        user_code_id: UID,
+        inputs: Dict,
+        node_name: str,
+    ) -> Result[Ok, Err]:
+        user_code_service = context.node.get_service("usercodeservice")
+        action_service = context.node.get_service("actionservice")
+        user_code = user_code_service.get_by_uid(context, uid=user_code_id)
+
+        res = user_code.status.mutate(
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/task/oblv_service.py:367]: Should this be a `ObjectMutation` change which gets applied to the `user_code` variable?

File: packages/syft/src/syft/core/node/new/user_code.py [hunk 5]
```diff
@@ -43,11 +55,16 @@ class InputPolicy(SyftObject):
     __canonical_name__ = "InputPolicy"
     __version__ = SYFT_OBJECT_VERSION_1
 
+    # relative
+    from .api import UserNodeView
+
     id: UID
-    inputs: Dict[str, Any]
+    inputs: Dict[UserNodeView, Any]
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/user_code.py:62]: The input policy key is the string of the kwarg name when invoking the function. What is the intention of using UserNodeView here instead?

File: packages/syft/src/syft/core/node/new/user_code.py [hunk 6]
```diff
@@ -252,7 +397,8 @@ class UserCode(SyftObject):
     user_unique_func_name: str
     code_hash: str
     signature: inspect.Signature
-    status: UserCodeStatus = UserCodeStatus.SUBMITTED
+    status: UserCodeStatusContext
+    enclave_metadata: Optional[EnclaveMetadata] = None
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/user_code.py:401]: This is enclave specific, i think we should consider a different or more generic way to store this.

File: packages/syft/src/syft/core/node/new/user_code.py [hunk 7]
```diff
@@ -252,7 +397,8 @@ class UserCode(SyftObject):
     user_unique_func_name: str
     code_hash: str
     signature: inspect.Signature
-    status: UserCodeStatus = UserCodeStatus.SUBMITTED
+    status: UserCodeStatusContext
```
Comment (madhavajay) [inline packages/syft/src/syft/core/node/new/user_code.py:400]: Why has the UserCodeStatus changed to `UserCodeStatusContext`? Is this for multiple permissions?

File: packages/syft/src/syft/core/node/new/credentials.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/credentials.py:46]: This is mainly for making the repr, short for a nice UI, maybe I will add it later in the UserNodeView class, I will revert it here now

File: packages/syft/src/syft/core/node/new/task/oblv_service.py (hunk 2 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/task/oblv_service.py:89]: These code paths , are used only when we use real enclave deployment in the cloud, 
We have a flag in `constants.py`
LOCAL_MODE, which switches between real and locally emulated enclave deployments, 
In Locally emulated enclave deployments we have to manually launch a FastAPI server

File: packages/syft/src/syft/core/node/new/task/oblv_service.py (hunk 3 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/task/oblv_service.py:56]: meaning subclassing from AnyActionObject?

File: packages/syft/src/syft/core/node/new/task/oblv_service.py (hunk 4 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/task/oblv_service.py:367]: Here the status object is a `UserCodeStatusContext` class  which is context aware on how to mutate permissions when multiple parties are involved

File: packages/syft/src/syft/core/node/new/user_code.py (hunk 5 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/user_code.py:62]: Currently we modify the input notation to also include the party it originated from.....

Assume we performing computation in Canada domain , we require access to the variable `data_1` in the data set

In the previous notations

```py

inputs = {"data_1" : UID() }
```

In the new notation:

```py
inputs = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : {"data_1" : UID() } }
```

Sort of like a nested dict, which maps the data from the parties it originated from , 

Consider two domains `Canada` , `Italy`
```py
inputs = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : {"data_1" : UID() },
                 UserNodeView(name="Italy", verify_key= canada_domain.verify_key) : {"data_2" : UID() },
 }
```

File: packages/syft/src/syft/core/node/new/user_code.py (hunk 6 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/user_code.py:401]: The is mainly set , when we work with an Enclave Client, 
and is set to None when we work with Domain Client

I mainly  to wanted to reduce different code paths and optimise UI flow,
Let me if you have any other idea 🙂 , we could modify it

File: packages/syft/src/syft/core/node/new/user_code.py (hunk 7 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/core/node/new/user_code.py:400]: Yep, this for multiple permissions

Each UserCodeStatusContext contains a dict, for the parties it requires approval from....

For one domain scenario:

```py
status = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : UserCodeStatus.SUBMITTED,
 }
```

For enclave scenario:

```py
status = {UserNodeView(name="Canada", verify_key= canada_domain.verify_key) : UserCodeStatus.SUBMITTED,
                  UserNodeView(name="Italy", verify_key= italy_domain.verify_key) : UserCodeStatus.SUBMITTED,
 }
```

**Review by madhavajay — APPROVED** [review]
🙌 Excellent work @rasswanth-s this is an amazing set of functionality which was demoed so well.

**Pull request merged**